### ISSUE TITLE
Convert quantized model to full int model

### ISSUE THREAD

**Issue by Narduzzi** [tensorflow/model-optimization#859]
**System information**

- TensorFlow version: 2.6.0
- Are you willing to contribute it (Yes/No): Yes

**Motivation**

It would be great to have a function that converts the model to full-int representation after quantization, in order to export the weights as integers.

**Describe the feature**

The pseudo-code would be: 
```python3
...
model = MyModel()
# quantize and train the model
quantized_model = tfmot.quantization.keras.quantize_model(model)
quantized_model.compile(model.optimizer, loss=model.loss, metrics=["accuracy"])
quantized_model.fit(x_train,y_train, validation_data=(x_test,y_test), epochs=1)

int_model = quantized_model.to_int() #converts model to integer weights and biases

for l in int_model.layers:
      print(l.get_weights()) # the weights are printed in float32, but rounded as integers, as computed by the QAT

```

**Describe how the feature helps achieve the use case**

I achieved your use case with the following code snippet. However, this was more difficult than it should be because of management of bias quantization, and accuracy is not good:

```python3
def dequantize(quantize_and_dequantized, min_var, max_var, quantizer):
    
    quantized_min = -(1 << (quantizer.num_bits - 1))
    scale = min_var / quantized_min
    quantized = tf.math.round(quantize_and_dequantized / scale).numpy().astype(np.int8)
    return quantized

def to_int_weights(model_arch, q_model, map_w={}):
    qweights = {}
    for layer in q_model.layers:

        if hasattr(layer, '_weight_vars'):
            weights = layer.get_weights()
            w = []
            b = []
            if len(weights)==7:
                b,w, _,_,_,_,_ = layer.get_weights()
            elif len(weights)==6: # no bias in layer
                w, _,_,_,_,_ = layer.get_weights()
                b=[]
                
            b = tf.convert_to_tensor(b)
            w = tf.convert_to_tensor(w)

            for weight, quantizer, quantizer_vars in layer._weight_vars:
                print(weight.shape)
                quantized_and_dequantized_w = quantizer(w, training=False, weights=quantizer_vars)
                quantized_and_dequantized_b = quantizer(b, training=False, weights=quantizer_vars)
                min_var = quantizer_vars['min_var']
                max_var = quantizer_vars['max_var']
                quantized_w = dequantize(quantized_and_dequantized_w, min_var, max_var, quantizer)
                quantized_b = dequantize(quantized_and_dequantized_b, min_var, max_var, quantizer)
                qweights[layer.name.replace("quant_","")] = {"w":quantized_w,
                                                            "b":quantized_b}    

    for l in model_arch.layers:
        w = l.get_weights()
        
        if l.name in map_w:
            corresponding_layer_name = map_w[l.name]
        else:
            corresponding_layer_name = l.name

            if hasattr(l, "kernel"):
                qw = qweights[corresponding_layer_name]
                w = qw["w"]
                b = qw["b"]
                if l.use_bias:
                    l.set_weights([w,b])
                else:
                    l.set_weights([w])

    return model_arch

# define architecture
model = ...
model_architecture = tf.keras.models.clone_model(model)
model_architecture.compile(loss=model.loss, metrics=["accuracy"])

# quantize and train
quantized_model = tfmot.quantization.keras.quantize_model(model)
quantized_model.compile(model.optimizer, loss=model.loss, metrics=["accuracy"])
quantized_model.fit(x_train,y_train, validation_data=(x_test,y_test), epochs=1)
quantized_model_accuracy = quantized_model.evaluate(x_test,y_test)[1]

# transfer quantized weights to model_architecture
int_model = to_int_weights(model_architecture, quantized_model)

for l in quantized_model.layers:
      print(l.get_weights()) # [[0.0245, -0.1245, ...]]
      
for l in int_model.layers:
      print(l.get_weights()) # [[2, -3, 5, ...]]
```

**Comment by daverim** [comment]
We do have vague future plans to support "materialization" of quantized weights but not any time soon -- in the meantime, your tool looks like it is mostly working, but you will need to ensure that the output quantization is also preserved to ensure model quality. Since we don't have all the necessary functionality in TF as xhark suggested, this would only really work for weight-only quantization.
